packages/common-db/src/lib/common/tools.ts

```diff
@@ -393,37 +423,58 @@ function findPath(path: string, url: string): string {
     }
 }
 
-function getMac(callback: (e?: Error | null, mac?: string) => void): void {
+/**
+ * Get MAC address of this host
+ */
+async function getMac(): Promise<string> {
     const macRegex = /(?:[a-z0-9]{2}[:-]){5}[a-z0-9]{2}/gi;
     const zeroRegex = /(?:[0]{2}[:-]){5}[0]{2}/;
     const command = process.platform.indexOf('win') === 0 ? 'getmac' : 'ifconfig || ip link';
 
-    exec(command, { windowsHide: true }, (err, stdout, _stderr) => {
-        if (err) {
-            callback(err);
-        } else {
-            let macAddress;
-            let match;
-            let result = null;
-
-            while (true) {
-                match = macRegex.exec(stdout);
-                if (!match) {
-                    break;
-                }
-                macAddress = match[0];
-                if (!zeroRegex.test(macAddress) && !result) {
-                    result = macAddress;
-                }
-            }
+    const { stdout, stderr } = await execAsync(command);
```

Comment:
no tryCatch? I think with callback it would call callback with err??

Reply:
In the end we catch it all on the outside

https://github.com/ioBroker/ioBroker.js-controller/blob/035a7b76ae8a84207fe9e530315365217c330963/packages/controller/src/main.ts#L2197-L2201